packages/amplify_datastore/ios/Classes/types/model/FlutterAuthRule.swift

```diff
@@ -0,0 +1,95 @@
+/*
+* Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+import Foundation
+import Amplify
+
+public struct FlutterAuthRule {
+    
+    private let ownerField : String?
+    private let identityClaim : String?
+    private let groupClaim : String?
+    private let groups : [String]?
+    private let groupsField : String?
+    private var authStrategy : String
+    private var operations : [String]?
+    
+    init(serializedData: [String: Any]) throws {
+
+        self.ownerField = serializedData["ownerField"] as? String
+
+        self.identityClaim = serializedData["identityClaim"] as? String
+        
+        self.groupClaim = serializedData["groupClaim"] as? String
+        
+        self.groups = serializedData["groups"] as? [String]
+        
+        self.groupsField = serializedData["groupsField"] as? String
+
+        guard let authStrategy = serializedData["authStrategy"] as? String
```

Comment:
Usually guard statements are kept at the top of the method. So to return early if fails.